jenkins-github-maven/Jenkinsfile

```diff
@@ -0,0 +1,226 @@
+@NonCPS
+def cancelPreviousBuilds() {
+    def jobName = env.JOB_NAME
+    def buildNumber = env.BUILD_NUMBER.toInteger()
+    def currentJob = Jenkins.instance.getItemByFullName(jobName)
+    println("This is: Job name #${jobName}, Build number #${buildNumber}, Current job #${currentJob}")
+
+    for(def build : currentJob.builds) {
+        println("Checking build #${build}")
+        def exec = build.getExecutor()
+        if(build.isBuilding() && build.number.toInteger() < buildNumber && exec != null) {
+            exec.interrupt(Result.ABORTED, new CauseOfInterruption.UserInterruption("Job aborted by later build #${buildNumber}"))
+            println("This build #${buildNumber} stopped build #${build}")
+        }
+    }
+}
+
+pipeline {
+    agent any
+
+    /*
+    This is here so that if the user tests fail, we don't keep analysing with Diffblue
+    before failing the job
+    */
+    options {
+        parallelsAlwaysFailFast()
+    }
+
+    /*
+    Modify these as required
+    */
+    environment {
+       // Key for Github from credentials set up on Jenkins server
+       SSH_KEY = credentials('SSH_KEY_PIGGYMETRICS')
```

Comment:
Is it your intention to keep PiggyMetrics-related stuff in here? :pig:

Reply:
For now, yes.